### PR TITLE
Assert false in IsApplicable method of TransformationAccessChain

### PR DESCRIPTION
**Pull request by stefanomil**
… if it is invalid (with null or undefined pointer).

In this case, assert false instead of returning false so that the fuzzer fails.

(See issue #3185 )

### CODE REVIEW

File: source/fuzz/transformation_access_chain.cpp [hunk 1]
```diff
@@ -74,10 +74,7 @@ bool TransformationAccessChain::IsApplicable(
   switch (pointer->opcode()) {
     case SpvOpConstantNull:
     case SpvOpUndef:
-      // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3185): When
-      //  fuzzing for real we would like an 'assert(false)' here.  But we also
-      //  want to be able to write negative unit tests.
-      return false;
+      assert(false && "Null or undefined pointer");
```
Comment (afd) [inline source/fuzz/transformation_access_chain.cpp:77]: Please change message to: "Access chains should not be created from null/undefined pointers".

And please add "return false" after the assertion.  That way in release mode - when there are no assertions - the function will return false, which is the right thing to do.

File: test/fuzz/transformation_access_chain_test.cpp [hunk 2]
```diff
@@ -184,14 +184,16 @@ TEST(TransformationAccessChainTest, BasicTest) {
                    .IsApplicable(context.get(), transformation_context));
 
   // Bad: pointer is null
-  ASSERT_FALSE(TransformationAccessChain(
+  ASSERT_DEATH(TransformationAccessChain(
                    100, 45, {80}, MakeInstructionDescriptor(24, SpvOpLoad, 0))
```
Comment (afd) [inline test/fuzz/transformation_access_chain_test.cpp:188]: Uses of ASSERT_DEATH need to be wrapped in:

```
#ifndef NDEBUG
// comment
ASSERT_DEATH(...);
#endif
```

See transformation_permute_phi_operands_test.cpp for an example.

This is to keep release compilers happy - death tests don't work in release mode.

Comment (stefanomil) [reply test/fuzz/transformation_access_chain_test.cpp]: Done

File: source/fuzz/transformation_access_chain.cpp (hunk 1 above)
Comment (stefanomil) [reply source/fuzz/transformation_access_chain.cpp]: Done

**Pull request merged**